### PR TITLE
Only run JIT formatting job if there are JIT changes

### PR DESCRIPTION
**Pull request by BruceForstall**

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet/jit-contrib

File: eng/pipelines/common/evaluate-default-paths.yml [hunk 1]
```diff
@@ -99,6 +99,9 @@ jobs:
       - src/coreclr/vm/*
       exclude:
       - '*'
+    - subset: coreclr_jit
+      include:
+      - src/coreclr/jit/*
```
Comment (BruceForstall) [inline eng/pipelines/common/evaluate-default-paths.yml:104]: I presume this means all files in the `jit` directory and all subdirectories of the `jit` directory?

Comment (kunalspathak) [reply eng/pipelines/common/evaluate-default-paths.yml:104]: looks like it.

**Review by kunalspathak — APPROVED** [review]

**Comment by danmoseley** [comment]
This looks right to me but someone like @safern should OK it.

File: eng/pipelines/common/evaluate-default-paths.yml (hunk 1 above)
Comment (safern) [reply eng/pipelines/common/evaluate-default-paths.yml:104]: Yes. If you want to know more about how this works, here is the summary of how it works: https://github.com/dotnet/runtime/blob/d471a03d2c5cebfad2311a3d0408248956d0e683/eng/pipelines/evaluate-changed-paths.sh#L2-L17

And this is the logic that takes care of that:
https://github.com/dotnet/runtime/blob/d471a03d2c5cebfad2311a3d0408248956d0e683/eng/pipelines/evaluate-changed-paths.sh#L132-L183

**Review by safern — APPROVED** [review]

**Pull request merged**